go/worker/compute/executor/committee/node.go

```diff
@@ -1291,6 +1318,90 @@ func (n *Node) handleExternalBatchLocked(batch *unresolvedBatch, hdr block.Heade
 	return nil
 }
 
+func (n *Node) handleHostRtEvent(ev *host.Event) {
+	switch {
+	case ev.Started != nil:
+		// We are now able to service requests for this runtime.
+		n.runtimeVersion = ev.Started.Version
+
+		n.roleProvider.SetAvailable(func(nd *node.Node) error {
+			rt := nd.AddOrUpdateRuntime(n.commonNode.Runtime.ID())
+			rt.Version = n.runtimeVersion
+			rt.Capabilities.TEE = ev.Started.CapabilityTEE
+			return nil
+		})
+	case ev.Updated != nil:
+		// Update runtime capabilities.
+		n.roleProvider.SetAvailable(func(nd *node.Node) error {
+			rt := nd.AddOrUpdateRuntime(n.commonNode.Runtime.ID())
+			rt.Version = n.runtimeVersion
+			rt.Capabilities.TEE = ev.Updated.CapabilityTEE
+			return nil
+		})
+	case ev.FailedToStart != nil, ev.Stopped != nil:
+		// Runtime failed to start or was stopped -- we can no longer service requests.
+		n.roleProvider.SetUnavailable()
+	default:
+		// Unknown event.
+		n.logger.Warn("unknown worker event",
+			"ev", ev,
+		)
+	}
+}
+
+func (n *Node) handleProcessedBatch(batch *processedBatch) {
```

Comment:
Could you refactor this function a bit? Two things:
* The state staleness checks are the same between the two cases (failed batch/successful batch) which is currently handled in two inline functions, so no need to duplicate them.
* These inline functions are not needed here (they were needed before as this was done inside a case, but now you already have a function), you can move locking/defer one level up.